sci-physics/hepmc/hepmc-3.3.0-r1.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 2022-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+inherit fortran-2 cmake python-single-r1
+
+MYP=HepMC3-${PV}
+
+DESCRIPTION="Event Record for Monte Carlo Generators"
+HOMEPAGE="https://hepmc.web.cern.ch/hepmc/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.cern.ch/hepmc/HepMC3"
+else
+	SRC_URI="https://hepmc.web.cern.ch/hepmc/releases/${MYP}.tar.gz"
+	S="${WORKDIR}/${MYP}"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3+"
+SLOT="3"
+IUSE="doc test examples python root static-libs"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="
+	python? (
+		${PYTHON_DEPS}
+		$(python_gen_cond_dep '
+			dev-python/numpy[${PYTHON_USEDEP}]
+		')
+	)
+"
+DEPEND="${RDEPEND}"
+# Automagic compression dependencies in test and example.
+# https://gitlab.cern.ch/hepmc/HepMC3/-/issues/99
+# For now we install all of them...
+BDEPEND="
+	root? ( sci-physics/root:= )
+	doc? (
+		app-text/doxygen[dot]
+		dev-texlive/texlive-latex
+		dev-texlive/texlive-latexextra
+		dev-texlive/texlive-latexrecommended
+	)
+	test? (
+		sys-libs/zlib
+		app-arch/xz-utils
+		app-arch/bzip2
+		app-arch/zstd
+	)
+	examples? (
+		sys-libs/zlib
+		app-arch/xz-utils
+		app-arch/bzip2
+		app-arch/zstd
+	)
+"
+
+src_configure() {
+	local mycmakeargs=(
```

Comment:
We're losing `filter-lto` here, is that intentional?

Reply:
Ah, I think this branch does not include the hepmc lto filter and a rebase won't add it because it a new file... I guess the safe way is to not have/develop on multiple branches at once. 

Or is there a better way to avoid this kind of errors (i.e. someone else modifies 3.3.0 while I create my changes in 3.3.0-r1)? Because with many people working on the same package in parallel manual checking for changes does not scale.

Reply:
Not sure if there is a better way. What I do is check the diff of the new ebuild versus the latest and 9999 in `::gentoo`  before merging, that's how I caught the missing `filter-lto`.

Reply:
The `9999` was automatically merged, but the `-r1` needed to be done manually. Ideally, I modify `9999` and copy that, thereby keeping `9999` also in working state.